### PR TITLE
[IRGen] Don't compute witness table layout just to form existentials

### PR DESCRIPTION
**Pull request by jrose-apple**
We weren't using this information, and trying to access it runs into trouble when the protocol is in another file.

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
@swift-ci Please test source compatibility

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test compiler performance

File: lib/IRGen/GenExistential.cpp [hunk 1]
```diff
@@ -162,49 +163,11 @@ namespace {
...
-    /// Given the witness table vector from an existential object, find the
-    /// witness table corresponding to the given protocol.
-    llvm::Value *findWitnessTable(IRGenFunction &IGF,
-                                  ArrayRef<llvm::Value *> witnesses,
-                                  ProtocolDecl *protocol) const {
-      return emitImpliedWitnessTableRef(IGF, getStoredProtocols(), protocol,
-        [&](unsigned originIndex) {
-          return witnesses[originIndex];
-        });
-    }
```
Comment (rjmccall) [inline lib/IRGen/GenExistential.cpp]: Was this all dead?  Unfortunate.

**Review by rjmccall — APPROVED** [review]
I assume the problem is that building an existential type alone doesn't validate the protocol enough to perform a layout of it?

Anyway, LGTM.

**Comment by jrose-apple** [comment]
Yep, that's correct. And yeah, I don't know if the dead code was *never* used or only *recently* unused, but that was the only code that wanted the ProtocolInfo.

**Comment by slavapestov** [comment]
Nice fix!

**Comment by jrose-apple** [comment]
Restarting tests due to fixing 80-column violations.

[Compiler perf](
https://ci.swift.org/job/swift-PR-compiler-performance-smoke-test-macOS/313/)
[Source compat](
https://ci.swift.org/job/swift-PR-source-compat-suite/1672/)
[Linux full](
https://ci.swift.org/job/swift-PR-Linux/6778/)
[macOS full](
https://ci.swift.org/job/swift-PR-osx/6772/)

@swift-ci Please smoke test

**Pull request merged**